File: src/shape/shape_subscriber.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"
#include "ros_basics_tutorials/Shape.h"

void chatterCallback(const ros_basics_tutorials::Shape::ConstPtr &msg)
{
    ROS_INFO("[Listener] I heard sensor %s with id %d, temp=%f, hum=%f\n", msg->name.c_str(), msg->id, msg->height, msg->width);
}
int main(int argc, char **argv)
{
    ros::init(argc, argv, "shape_subscriber");
    ros::NodeHandle nh;
    ros::Subscriber chatter_subscriber = nh.subscribe("shape_info", 1000, chatterCallback);
    ros::spin();

    return 0;
}

File: src/shape/shape_subscriber.py
#!/usr/bin/env python3
import rospy
from std_msgs.msg import String
from ros_basics_tutorials.msg import Shape


def chatter_callback(message):
    rospy.loginfo("I heard: (%d, %s, %.2f, %.2f)", message.id,
                  message.name, message.height, message.width)


def listener():
    rospy.init_node('shape_subscriber_py', anonymous=True)

    rospy.Subscriber('shape_info', Shape, chatter_callback)

    # To start listen, it will keep listening
    rospy.spin()


if __name__ == '__main__':
    try:
        listener()
    except rospy.ROSInterruptException:
        pass
